src/mfakto.cpp

```diff
@@ -327,17 +327,20 @@ int init_CL(int num_streams, cl_int *devnumber)
     else for(i=0; i < numplatforms; i++) // autoselect: search for AMD
     {
       char buf[128] = {0};
+      cl_uint device_count;
+      int is_amd;
+
       status = clGetPlatformInfo(platformlist[i], CL_PLATFORM_VENDOR,
                         sizeof(buf), buf, NULL);
       if(status != CL_SUCCESS)
       {
         std::cerr << "Error " << status << " (" << ClErrorString(status) << "): clGetPlatformInfo(VENDOR)\n";
         return 1;
       }
-      platform = platformlist[i];  // use any platform, but ...
-      if(strcmp(buf, "Advanced Micro Devices, Inc.") == 0)
-      {
-        break;  // ... prefer AMD, otherwise use the last one
+      is_amd = strcmp(buf, "Advanced Micro Devices, Inc.") == 0;
```

Comment:
This needs to be here, as enabling `DETAILED_INFO` would rewrite `buf` with different data.